broker/core/src/mysql_connection.cc

```diff
@@ -674,18 +752,39 @@ void mysql_connection::_run() {
     _update_stats();
     lck.unlock();
 
+    std::chrono::system_clock::time_point last_commit =
+        std::chrono::system_clock::now();
+
     std::unique_lock<std::mutex> lock(_tasks_m);
+    bool reconnect_failed_logged = false;
     while (_state == running || !_tasks_list.empty()) {
-      std::list<std::unique_ptr<database::mysql_task>> tasks_list;
+      if (_error.is_active()) {
+        lock.unlock();
+        if (!_try_to_reconnect()) {
+          if (!reconnect_failed_logged) {
+            SPDLOG_LOGGER_ERROR(log_v2::sql(), "SQL: Reconnection failed.");
+            reconnect_failed_logged = true;
+          }
+          std::this_thread::sleep_for(std::chrono::seconds(10));
+          lock.lock();
+          continue;
+        } else {
+          reconnect_failed_logged = false;
+        }
+        lock.lock();
+      }
+      mysql_task* to_execute;
       if (!_tasks_list.empty()) {
-        std::swap(_tasks_list, tasks_list);
```

Comment:
I would like we continue to use a local tasks_list with a call to swap to avoid all the locks/unlocks needed otherwise.